### PR TITLE
feat(bot): auto update links of moved files

### PR DESCRIPTION
**Pull request by OnkarRuikar**
Every night the bot will create a PR to update all the links of moved files.

- Note: A demo PR, https://github.com/mdn/content/pull/31090, was created using the code and has already been merged. The PR has fixed all the previous redirect issues in the content.

### CODE REVIEW

**Comment by Josh-Cena** [comment]
Can this be part of the Yari CLI instead? It is my opinion that things that can be fixed locally should be fixed locally instead of waiting for another cleanup PR. Again PRs and commits are not zero-cost. We have formatting/linting daily fix because not everyone authors content locally, but everyone is and should be using the `move` CLI to move files.

**Comment by OnkarRuikar** [comment]
> Can this be part of the Yari CLI instead? It is my opinion that things that can be fixed locally should be fixed locally, instead of waiting for another cleanup PR. Again, PRs and commits are not zero-cost. We have formatting/linting daily fixes, because not everyone authors content locally, but everyone is and should be using the `move` CLI to move files.

- I did submit a yari PR 9 months ago which does exactly what you are saying: https://github.com/mdn/yari/pull/8493 .
Multiple review & merge requests were made. Multiple rebases and conflict resolutions also happened. The PRs came up in the last community meeting as well. :rage: In OWD meeting last week we decided to have the implementation as an auto cleanup bot on content side.
If you could get the yari PR merged then it would be great. Meanwhile there is no harm in keeping this.

**Comment by Josh-Cena** [comment]
Ugh, I keep forgetting how painful it is to get things merged in Yari. I feel you. No objection from me, then.

**Comment by OnkarRuikar** [comment]
@teoli2003 ping

File: scripts/update-moved-file-links.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+      console.warn("No redirect entry found for: ", tuple.from);
+    }),
+  )
+).filter((e) => !!e);
+
+console.log(`Number of moved files to consider: ${movedFiles.length}`);
+
+let totalNo = 0,
+  updatedNo = 0;
```
Comment (teoli2003) [inline scripts/update-moved-file-links.js:105]: Nit:

I prefer this syntax:
```suggestion
let totalNo = 0;
let updatedNo = 0;
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
One nit.

I think we should then try this, keeping in mind that once Yari's PR will be merged we will undo this. (But this is an annoying problem, so worth solving it in the meantime)

**Review by teoli2003 — APPROVED** [review]
We should go ahead with this:
- this can't break mdn/content (and the Firefox release date was yesterday, so it can't be a problem there)
- it is easy to revert in case of problem (the modified files aren't modified every day)
- it is a band-aid

Let's merge this.

**Pull request merged**